### ISSUE TITLE
Check for disambiguation page in wiki info

### ISSUE THREAD

**Issue by boothym** [openstreetmap/iD#4414]
Especially for place names, sometimes the first result in the wikipedia drop-down is actually a disambiguation page. Is there any way to detect this and warn the user so that the wrong info doesn't get added?

**Comment by bhousel** [comment]
@1ec5 can I assign this to you?

**Comment by 1ec5** [comment]
Yeah, although I’m not entirely clear on how to fix this issue at the moment. The Wikipedia field appears to perform a very different API action based on the input length. `action=query` [does have a way](https://www.mediawiki.org/wiki/Extension:Disambiguator#API_usage) to check whether a specific page is a disambiguation page, but I [can’t get it to work](https://en.wikipedia.org/wiki/Special:ApiSandbox#action=query&format=json&prop=pageprops&list=search&ppprop=disambiguation&srsearch=beijing%20%28disambiguation%29&srinfo=totalhits|suggestion|rewrittenquery&srprop=size|wordcount|timestamp|snippet) with `list=search`. There’s [no such option](https://en.wikipedia.org/wiki/Special:ApiSandbox#action=opensearch&format=json&search=Beijing) for `action=opensearch`, although this would be a good opportunity to switch to `action=query&generator=prefixsearch`, which can pull in Wikidata-powered article summaries in the dropdown suggestions along the lines of [what the www.wikipedia.org portal does](https://github.com/wikimedia/portals/blob/bdb3edc60de86bb29c98d69a8061fed94723294e/dev/wikipedia.org/assets/js/wm-typeahead.js#L182-L197).

**Comment by bhousel** [comment]
#5647 solves this

**Comment by matkoniecz** [comment]
It appears to be not fixed or regressed. After selecting Polish Wikipedia ("polski") in WIkipedia article adding menu and typing "Kalenic" I get "Kalenica" disambiguation page as a suggestion.

And #5647 seems to be about integration with OSM Wiki Wikidata, not with Wikidata ("Takes data directly from the Wikibase data items (OSM Wiki)")

**Comment by quincylvania** [comment]
> It appears to be not fixed or regressed.

Verified.

**Comment by 1ec5** [comment]
> `action=query` [does have a way](https://www.mediawiki.org/wiki/Extension:Disambiguator#API_usage) to check whether a specific page is a disambiguation page, but I [can’t get it to work](https://en.wikipedia.org/wiki/Special:ApiSandbox#action=query&format=json&prop=pageprops&list=search&ppprop=disambiguation&srsearch=beijing%20%28disambiguation%29&srinfo=totalhits|suggestion|rewrittenquery&srprop=size|wordcount|timestamp|snippet) with `list=search`.

That’s because `prop=pageprops` (and thus `ppprop=disambiguation`) only works when setting `titles=` to specific page titles, not when searching with `srsearch=`. Adding an `srprop` might be tracked in [T18449](https://phabricator.wikimedia.org/T18449).

So I think filtering out disambiguation pages would require an extra call to the MediaWiki API for each search result, which would probably not be worth the trouble. On the other hand, if there’s a way to validate a field value as soon as it’s entered (or selected from the dropdown), there are multiple ways to check an individual page’s disambiguation status, such as using `ppprop=disambiguation`.

> And #5647 seems to be about integration with OSM Wiki Wikidata, not with Wikidata ("Takes data directly from the Wikibase data items (OSM Wiki)")

Perhaps the thought was that, by automatically populating the Wikidata field with the linked Wikidata item’s description, the user would get a chance to see “Wikipedia disambiguation page” in that field and realize they’ve chosen the wrong Wikipedia article. But I agree that it isn’t quite a fix either.
